roles/spark/tasks/main.yml

```diff
@@ -0,0 +1,248 @@
+    mode: '0644'
+
+- name: create spark master systemd service file
+  copy:
+    dest: /etc/systemd/system/spark-master.service
+    content: |
+      [Unit]
+      Description=Apache Spark Master
+      After=network.target
+
+      [Service]
+      Type=forking
+      User=root
+      Group=root
+      ExecStart=/opt/spark/sbin/start-master.sh
+      ExecStop=/opt/spark/sbin/stop-master.sh
+      Restart=on-failure
+
+      [Install]
+      WantedBy=multi-user.target
+  delegate_to: "{{ groups['spark_master'][0] }}"
+  run_once: true
+
+- name: enable spark master
+  systemd:
+    name: spark-master
+    enabled: yes
+  delegate_to: "{{ groups['spark_master'][0] }}"
+  run_once: true
+
+- name: create spark worker systemd service file
+  copy:
+    dest: /etc/systemd/system/spark-workers.service
+    content: |
+      [Unit]
+      Description=Apache Spark Workers
+      After=network.target
+
+      [Service]
+      Type=simple
+      User=root
+      Group=root
+      ExecStart=/opt/spark/sbin/start-workers.sh spark://{{ spark_data.server[0].instance.ip_priv1 }}:7077
+      ExecStop=/opt/spark/sbin/stop-workers.sh
+      Restart=on-failure
+      RemainAfterExit=yes
+      
+      [Install]
+      WantedBy=multi-user.target
+  delegate_to: "{{ groups['spark_master'][0] }}"
+  run_once: true
+
+- name: enable spark workers
+  systemd:
+    name: spark-workers
+    enabled: yes
+  delegate_to: "{{ groups['spark_master'][0] }}"
+  run_once: true
+
+- name: install scala
+  apt:
+    name: scala
+    state: present
+    update_cache: yes
+
+- name: add spark environment variables to .bashrc
+  lineinfile:
+    path: "{{ ansible_env.HOME }}/.bashrc"
+    create: yes
+    line: "{{ item }}"
+    state: present
+  with_items:
+    - 'export SPARK_HOME=/opt/spark'
+    - 'export PATH=$PATH:$SPARK_HOME/bin:$SPARK_HOME/sbin'
+
+- name: source .bashrc
+  shell: source ~/.bashrc
+  args:
+    executable: /bin/bash
+
+- name: create spark-events directory
+  file:
+    path: /var/log/spark-events
+    state: directory
+    owner: root
+    group: root
+    mode: '0755'
+
+# create let's encrypt certs for master
+- name: create and install ssl certificates
+  import_tasks: ssl.yml
+  delegate_to: "{{ groups['spark_master'][0] }}"
+  run_once: true
+
+# instal nginx reverse proxy for authentication to master 
+- name: install nginx for reverse proxy auth
+  import_tasks: nginx.yml
+  delegate_to: "{{ groups['spark_master'][0] }}"
+  run_once: true
+
+- name: set up SSL certificate renewal cronjob
+  cron:
+    name: "Renew Let's Encrypt SSL certificates"
+    job: "/usr/bin/certbot renew --nginx --quiet --deploy-hook 'systemctl reload nginx'"
+    minute: "0"
+    hour: "3"
+    day: "1"
+    month: "*/3"
+    user: root
+  delegate_to: "{{ groups['spark_master'][0] }}"
+  run_once: true
+
+- name: add core-site.xml file
+  template:
+    src: core-site.xml.j2
+    dest: /opt/spark/conf/core-site.xml
+  delegate_to: "{{ groups['spark_master'][0] }}"
```

Comment:
This file should be in file and not templates. There are no variables being used.